lib/uri.js

```diff
@@ -90,6 +93,10 @@ URI.prototype.toString = function(options) {
                 // Omit optional segment & return
                 return uriStr;
             }
+        } else if (this._pathMetadata
+                && this._pathMetadata[i]
+                && this._pathMetadata[i] === '+') {
+            uriStr += '/' + utils.encodeReserved(segment);
```

Comment:
@gwicke While unreserved being ` unreserved     =  ALPHA / DIGIT / "-" / "." / "_" / "~"`, so we need to encode others, like spaces, or unicode etc.